File: Java+/Spring+/spring-java13-example/src/test/java/app/data/jpa/PersonRepositoryTest.java
package app.data.jpa;

import app.BaseJpaTest;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;

import java.util.Optional;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.contains;
import static org.hamcrest.Matchers.equalTo;
import static org.hamcrest.Matchers.notNullValue;
import static org.junit.jupiter.api.Assertions.assertTrue;


class PersonRepositoryTest extends BaseJpaTest {

    @Autowired
    private PersonRepository personRepository;

    @Test
    void findAll() {
        Person person1 = personRepository.save(new Person());
        Person person2 = personRepository.save(new Person());
        Iterable<Person> allPersons = personRepository.findAll();
        assertThat(allPersons, contains(person1, person2));
    }

    @Test
    void save() {
        Person origin = new Person();
        String name = "John";
        origin.setName(name);
        Person exp = personRepository.save(origin);
        assertThat(exp.getId(), notNullValue());
        assertThat(exp.getName(), equalTo(name));
        Optional<Person> optAct = personRepository.findById(exp.getId());
        assertTrue(optAct.isPresent());
        Person act = optAct.get();
        assertThat(act, equalTo(exp));
    }
}

File: Kotlin+/KotlinSpringExample/src/test/kotlin/app/data/jpa/PersonRepositoryTest.kt
package app.data.jpa

import app.BaseJpaTest
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired

internal class PersonRepositoryTest : BaseJpaTest() {
    @Autowired
    private lateinit var personRepository: PersonRepository

    @Test
    fun findAll() {
        val person1 = personRepository.save(Person())
        val person2 = personRepository.save(Person())
        val allPersons = personRepository.findAll()
        assertThat(allPersons).containsOnly(person1, person2)
    }

    @Test
    fun save() {
        val name = "John"
        val origin = Person(name = name)
        val exp = personRepository.save(origin)
        assertThat(exp.id).isNotNull()
        assertThat(exp.name).isEqualTo(name)
        assertThat(personRepository.findById(exp.id!!)).hasValue(exp)
    }
}